src/core/manager.h

```diff
@@ -277,6 +277,9 @@ struct Manager {
         and it is much simpler that way. */
         sd_bus_track *subscribed;
         char **deserialized_subscribed;
+        /* We'll send out legacy JobRemoved signals only if at least one subscriber asks for it by calling
+         * the legacy Subscribe() method. */
+        size_t n_legacy_subscribers;
```

Comment:
please just maintain two sd_bus_track objects: one for new-style and one for old-style subscribers. And send the JobRemoved if the old-style one is populated and JobRemoevedEx if the new-style one is populated.

Reply:
It's quite _a lot_ of churn to duplicate everything, are we sure it's worth it? It's just checked in one place after all

Reply:
there usually should not be duplicate entries. i.e. old clients in one, and new clients in the other. so you got the same number of entries either way. All you pay is that you allocate the track object twice. But that's cheap... in particular as we lazy-allocate the sd_bus_track object iirc, and thus if you have a fully new-style system the old object will never be allocated.

Reply:
I'm not worried about the runtime cost - by churn I meant code churn. Everything needs to be duplicated, and handled in tandem everywhere, save for one place - where the signal is sent. It's quite a lot of duplication.

Reply:
"everything"? it's not that much, is it?

Reply:
Everything to do with this queue, including serializing/deserializing, etc